rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -124,24 +137,38 @@
             @RequestParam(value = "name", required = false) String name,
             @RequestParam(value = "type", required = false) String projectType,
             @RequestParam(value = "group", required = false) String group,
-            @RequestParam(value = "tag", required = false) String tag) throws TException {
+            @RequestParam(value = "tag", required = false) String tag,
+            @RequestParam(value = "allDetails", required = false) boolean allDetails) throws TException {
         User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        Map<String, Project> mapOfProjects = new HashMap<>();
+        boolean isSearchByName = false;
```

Comment:
May be this can be `final boolean isSearchByName = name != null && !name.isEmpty()` and can be used in if clause and subsequently can be passed to the method.

Reply:
Changes done